felucca/backend/execution_manager.py

```diff
@@ -1,2 +1,143 @@
+import sys
+sys.path.append(r'./common')
+import os
+import argparse
+import shlex
+import tarfile
+import docker
+from common.singleton import Singleton
+from threading import Thread
+
+@Singleton
 class ExecutionManager(object):
-    pass
\ No newline at end of file
+    
```

Comment:
Add comments in class & methods as description.
Also add some comments for code blocks as explanation.

Reply:
get it